### PR TITLE
[SPARK-15177.1] [R] make SparkR model params and default values consistent with MLlib

### PR DESCRIPTION
**Pull request by mengxr**
## What changes were proposed in this pull request?

This PR is a subset of #13023 by @yanboliang to make SparkR model param names and default values consistent with MLlib. I tried to avoid other changes from #13023 to keep this PR minimal. I will send a follow-up PR to improve the documentation.

Main changes:
- `spark.glm`: epsilon -> tol, maxit -> maxIter
- `spark.kmeans`: default k -> 2, default maxIter -> 20, default initMode -> "k-means||"
- `spark.naiveBayes`: laplace -> smoothing, default 1.0
## How was this patch tested?

Existing unit tests.

### CODE REVIEW

**Comment by mengxr** [comment]
cc: @shivaram

File: R/pkg/R/mllib.R [hunk 1]
```diff
@@ -298,17 +296,17 @@ setMethod("summary", signature(object = "NaiveBayesModel"),
 #' @export
 #' @examples
 #' \dontrun{
-#' model <- spark.kmeans(data, ~ ., k=2, initMode="random")
+#' model <- spark.kmeans(data, ~ ., k = 4, initMode = "random")
 #' }
 #' @note spark.kmeans since 2.0.0
 setMethod("spark.kmeans", signature(data = "SparkDataFrame", formula = "formula"),
-          function(data, formula, k, maxIter = 10, initMode = c("random", "k-means||")) {
+          function(data, formula, k = 2, maxIter = 20, initMode = c("k-means||", "random")) {
```
Comment (shivaram) [inline R/pkg/R/mllib.R]: just to clarify - this initMode change wasn't present in #13023 -- Is this intended to match some Spark behavior ?

Comment (mengxr) [inline R/pkg/R/mllib.R]: Yes, the default `initMode` in MLlib is `k-means||` instead of `random`.

**Comment by shivaram** [comment]
Changes look fine given what was a part of #13023

File: R/pkg/R/mllib.R (hunk 1 above)
Comment (shivaram) [inline R/pkg/R/mllib.R]: Ah I see - change LGTM then

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
Merged into master and branch-2.0.

**Pull request closed without merging**